### ISSUE TITLE
Missing self_attn.q_norm.weight when merging olmo2

### ISSUE THREAD

**Issue by Tan-Hexiang** [arcee-ai/mergekit#602]
When  trying  to  merge several checkpoint of olmo2 (olmo2-7b-sft, olmo2-7b-dpo, olmo2-7b-instruct), the mergekit complete the merging process without any errors!  But the merged model is **missing some parameters** and will report an error when imported from from_pretrained.

The error information is shown as follows:
`Some weights of Olmo2ForCausalLM were not initialized from the model checkpoint at /cfs/hadoop-aipnlp/tanhexiang/merged_models/linear_all_6 and are newly initialized: ['model.layers.0.self_attn.k_norm.weight', 'model.layers.0.self_attn.q_norm.weight', 'model.layers.1.self_attn.k_norm.weight', 'model.layers.1.self_attn.q_norm.weight', 'model.layers.10.self_attn.k_norm.weight', 'model.layers.10.self_attn.q_norm.weight', 'model.layers.11.self_attn.k_norm.weight', 'model.layers.11.self_attn.q_norm.weight', 'model.layers.12.self_attn.k_norm.weight', 'model.layers.12.self_attn.q_norm.weight', 'model.layers.13.self_attn.k_norm.weight', 'model.layers.13.self_attn.q_norm.weight', 'model.layers.14.self_attn.k_norm.weight', 'model.layers.14.self_attn.q_norm.weight', 'model.layers.15.self_attn.k_norm.weight', 'model.layers.15.self_attn.q_norm.weight', 'model.layers.16.self_attn.k_norm.weight', 'model.layers.16.self_attn.q_norm.weight', 'model.layers.17.self_attn.k_norm.weight', 'model.layers.17.self_attn.q_norm.weight', 'model.layers.18.self_attn.k_norm.weight', 'model.layers.18.self_attn.q_norm.weight', 'model.layers.19.self_attn.k_norm.weight', 'model.layers.19.self_attn.q_norm.weight', 'model.layers.2.self_attn.k_norm.weight', 'model.layers.2.self_attn.q_norm.weight', 'model.layers.20.self_attn.k_norm.weight', 'model.layers.20.self_attn.q_norm.weight', 'model.layers.21.self_attn.k_norm.weight', 'model.layers.21.self_attn.q_norm.weight', 'model.layers.22.self_attn.k_norm.weight', 'model.layers.22.self_attn.q_norm.weight', 'model.layers.23.self_attn.k_norm.weight', 'model.layers.23.self_attn.q_norm.weight', 'model.layers.24.self_attn.k_norm.weight', 'model.layers.24.self_attn.q_norm.weight', 'model.layers.25.self_attn.k_norm.weight', 'model.layers.25.self_attn.q_norm.weight', 'model.layers.26.self_attn.k_norm.weight', 'model.layers.26.self_attn.q_norm.weight', 'model.layers.27.self_attn.k_norm.weight', 'model.layers.27.self_attn.q_norm.weight', 'model.layers.28.self_attn.k_norm.weight', 'model.layers.28.self_attn.q_norm.weight', 'model.layers.29.self_attn.k_norm.weight', 'model.layers.29.self_attn.q_norm.weight', 'model.layers.3.self_attn.k_norm.weight', 'model.layers.3.self_attn.q_norm.weight', 'model.layers.30.self_attn.k_norm.weight', 'model.layers.30.self_attn.q_norm.weight', 'model.layers.31.self_attn.k_norm.weight', 'model.layers.31.self_attn.q_norm.weight', 'model.layers.4.self_attn.k_norm.weight', 'model.layers.4.self_attn.q_norm.weight', 'model.layers.5.self_attn.k_norm.weight', 'model.layers.5.self_attn.q_norm.weight', 'model.layers.6.self_attn.k_norm.weight', 'model.layers.6.self_attn.q_norm.weight', 'model.layers.7.self_attn.k_norm.weight', 'model.layers.7.self_attn.q_norm.weight', 'model.layers.8.self_attn.k_norm.weight', 'model.layers.8.self_attn.q_norm.weight', 'model.layers.9.self_attn.k_norm.weight', 'model.layers.9.self_attn.q_norm.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.`

I have checked that all these parameters are in the source models and the names are same.

Here is the config file:
`models:
  - model: /....../OLMo-2-1124-7B-SFT
    parameters:
      weight: 1.0
  - model: /....../OLMo-2-1124-7B-DPO
    parameters:
      weight: 1.0
  - model: /....../OLMo-2-1124-7B-Instruct
    parameters:
      weight: 1.0
merge_method: linear
dtype: float16
`

I also tried the following config, resulting in the same problem:
`models:
  - model: /....../OLMo-2-1124-7B
    parameters:
      weight: 1.0
  - model: /....../OLMo-2-1124-7B-SFT
    parameters:
      weight: 1.0
  - model: /....../OLMo-2-1124-7B-DPO
    parameters:
      weight: 1.0
  - model: /....../OLMo-2-1124-7B-Instruct
    parameters:
      weight: 1.0
base_model: /....../OLMo-2-1124-7B-Instruct
merge_method: linear
dtype: float16
unweighted_parameters:
  - '.*_norm\\.weight'
  - '.*_norm\\.bias'`

The following is the complete environments (result of pip list):
`absl-py                            2.1.0
accelerate                         1.6.0
aiohappyeyeballs                   2.6.1
aiohttp                            3.11.14
aiosignal                          1.3.2
annotated-types                    0.7.0
anyio                              4.9.0
anytree                            2.12.1
argon2-cffi                        23.1.0
argon2-cffi-bindings               21.2.0
arrow                              1.3.0
asttokens                          3.0.0
async-lru                          2.0.5
attrs                              25.3.0
babel                              2.17.0
beautifulsoup4                     4.13.3
bert-score                         0.3.13
bioc                               2.1
bleach                             6.2.0
boto3                              1.37.15
botocore                           1.37.15
certifi                            2025.1.31
cffi                               1.17.1
charset-normalizer                 3.4.1
click                              8.2.1
cloudpickle                        3.1.1
comm                               0.2.2
conllu                             4.5.3
contourpy                          1.3.1
cycler                             0.12.1
datasets                           3.4.1
debugpy                            1.8.13
decorator                          5.2.1
defusedxml                         0.7.1
Deprecated                         1.2.18
dill                               0.3.8
diskcache                          5.6.3
distro                             1.9.0
docker-pycreds                     0.4.0
docopt                             0.6.2
editdistance                       0.8.1
einops                             0.8.1
executing                          2.2.0
faiss                              1.11.0
faiss-gpu-cu12                     1.11.0
fastapi                            0.115.11
fastjsonschema                     2.21.1
filelock                           3.18.0
flair                              0.15.1
fonttools                          4.56.0
fqdn                               1.5.1
frozenlist                         1.5.0
fsspec                             2024.12.0
ftfy                               6.3.1
gdown                              5.2.0
gguf                               0.10.0
gitdb                              4.0.12
GitPython                          3.1.44
gpytorch                           1.14
grpcio                             1.71.0
h11                                0.14.0
hf-xet                             1.1.3
httpcore                           1.0.7
httptools                          0.6.4
httpx                              0.28.1
huggingface-hub                    0.32.4
idna                               3.10
immutables                         0.21
importlib_metadata                 8.6.1
interegular                        0.3.3
intervaltree                       3.1.0
ipykernel                          6.29.5
ipython                            9.0.2
ipython_pygments_lexers            1.1.1
isoduration                        20.11.0
jaxtyping                          0.3.0
jedi                               0.19.2
jieba                              0.42.1
Jinja2                             3.1.6
jiter                              0.9.0
jmespath                           1.0.1
joblib                             1.4.2
json5                              0.10.0
jsonlines                          4.0.0
jsonpointer                        3.0.0
jsonschema                         4.23.0
jsonschema-specifications          2024.10.1
jupyter_client                     8.6.3
jupyter_core                       5.7.2
jupyter-events                     0.12.0
jupyter-lsp                        2.2.5
jupyter_server                     2.15.0
jupyter_server_terminals           0.5.3
jupyterlab                         4.3.6
jupyterlab_pygments                0.3.0
jupyterlab_server                  2.27.3
kiwisolver                         1.4.8
langdetect                         1.0.9
language_tool_python               2.9.0
lark                               1.2.2
lemminflect                        0.2.3
linear-operator                    0.6
llvmlite                           0.44.0
lm-format-enforcer                 0.10.6
lru-dict                           1.3.0
lxml                               5.3.1
Markdown                           3.7
MarkupSafe                         3.0.2
matplotlib                         3.7.5
matplotlib-inline                  0.1.7
mergekit                           0.1.3          /....../mergekit
mistral_common                     1.5.4
mistune                            3.1.3
more-itertools                     10.6.0
mpld3                              0.5.10
mpmath                             1.3.0
msgpack                            1.1.0
msgspec                            0.19.0
multidict                          6.2.0
multiprocess                       0.70.16
nbclient                           0.10.2
nbconvert                          7.16.6
nbformat                           5.10.4
nest-asyncio                       1.6.0
netcal                             1.3.6
networkx                           3.4.2
nltk                               3.8
notebook                           7.3.3
notebook_shim                      0.2.4
num2words                          0.5.14
numba                              0.61.0
numpy                              1.26.4
nvidia-cublas-cu12                 12.1.3.1
nvidia-cuda-cupti-cu12             12.1.105
nvidia-cuda-nvrtc-cu12             12.1.105
nvidia-cuda-runtime-cu12           12.1.105
nvidia-cudnn-cu12                  9.1.0.70
nvidia-cufft-cu12                  11.0.2.54
nvidia-curand-cu12                 10.3.2.106
nvidia-cusolver-cu12               11.4.5.107
nvidia-cusparse-cu12               12.1.0.106
nvidia-ml-py                       12.570.86
nvidia-nccl-cu12                   2.20.5
nvidia-nvjitlink-cu12              12.8.93
nvidia-nvtx-cu12                   12.1.105
nvitop                             1.4.2
openai                             1.66.5
opencv-python-headless             4.11.0.86
OpenHowNet                         2.0
opt_einsum                         3.4.0
outlines                           0.0.46
overrides                          7.7.0
packaging                          24.2
pandas                             2.2.3
pandocfilters                      1.5.1
parso                              0.8.4
partial-json-parser                0.2.1.1.post5
peft                               0.15.2
pexpect                            4.9.0
pillow                             11.1.0
pinyin                             0.4.0
pip                                24.0
platformdirs                       4.3.6
pptree                             3.1
prometheus_client                  0.21.1
prometheus-fastapi-instrumentator  7.0.2
prompt_toolkit                     3.0.50
propcache                          0.3.0
protobuf                           5.29.3
psutil                             7.0.0
ptyprocess                         0.7.0
pure_eval                          0.2.3
py-cpuinfo                         9.0.0
pyairports                         2.1.1
pyarrow                            19.0.1
pycountry                          24.6.1
pycparser                          2.22
pydantic                           2.10.6
pydantic_core                      2.27.2
Pygments                           2.19.1
pyparsing                          3.2.1
pyro-api                           0.1.2
pyro-ppl                           1.9.1
PySocks                            1.7.1
python-dateutil                    2.9.0.post0
python-dotenv                      1.0.1
python-json-logger                 3.3.0
pytorch_revgrad                    0.2.0
pytz                               2025.1
PyYAML                             6.0.2
pyzmq                              26.3.0
ray                                2.43.0
referencing                        0.36.2
regex                              2024.11.6
requests                           2.32.3
rfc3339-validator                  0.1.4
rfc3986-validator                  0.1.1
rpds-py                            0.23.1
s3transfer                         0.11.4
safetensors                        0.5.3
scikit-learn                       1.5.2
scipy                              1.15.2
seaborn                            0.13.2
segtok                             1.5.11
Send2Trash                         1.8.3
sentence-transformers              4.1.0
sentencepiece                      0.2.0
sentry-sdk                         2.23.1
setproctitle                       1.3.5
setuptools                         69.5.1
six                                1.17.0
smmap                              5.0.2
sniffio                            1.3.1
sortedcontainers                   2.4.0
soupsieve                          2.6
sqlitedict                         2.1.0
stack-data                         0.6.3
starlette                          0.46.1
sympy                              1.13.3
tabulate                           0.9.0
tensorboard                        2.19.0
tensorboard-data-server            0.7.2
terminado                          0.18.1
terminaltables                     3.1.10
textattack                         0.3.10
threadpoolctl                      3.6.0
tiktoken                           0.9.0
tikzplotlib                        0.9.8
tinycss2                           1.4.0
tokenizers                         0.21.1
toml                               0.10.2
torch                              2.4.0
torchvision                        0.19.0
tornado                            6.4.2
tqdm                               4.67.1
traitlets                          5.14.3
transformer-smaller-training-vocab 0.4.0
transformers                       4.48.0
triton                             3.0.0
types-python-dateutil              2.9.0.20241206
typing_extensions                  4.12.2
tzdata                             2025.1
uri-template                       1.3.0
urllib3                            2.3.0
uvicorn                            0.34.0
uvloop                             0.21.0
vllm                               0.6.3
wadler_lindig                      0.1.4
wandb                              0.19.8
watchfiles                         1.0.4
wcwidth                            0.2.13
webcolors                          24.11.1
webencodings                       0.5.1
websocket-client                   1.8.0
websockets                         15.0.1
Werkzeug                           3.1.3
wheel                              0.43.0
Wikipedia-API                      0.8.1
word2number                        1.1
wrapt                              1.17.2
xformers                           0.0.27.post2
xxhash                             3.5.0
yarl                               1.18.3
zipp                               3.21.0`
